core/src/marker/markerManager.cpp

```diff
@@ -14,15 +15,20 @@
 
 namespace Tangram {
 
+MarkerManager::MarkerManager() {}
+
+MarkerManager::~MarkerManager() {}
+
 void MarkerManager::setScene(std::shared_ptr<Scene> scene) {
 
     m_scene = scene;
     m_mapProjection = scene->mapProjection().get();
-    m_styleContext.initFunctions(*scene);
-    m_jsFnIndex = scene->functions().size();
 
+    m_styleContext = std::make_unique<StyleContext>();
+    m_styleContext->initFunctions(*scene);
 
     // Initialize StyleBuilders.
+    m_styleBuilders.clear();
```

Comment:
And when this gets included we can close https://github.com/tangrams/tangram-es/pull/1467 in favor of this PR.